demos/common/utilities/azure_trust_bundle_storage.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#include <stdint.h>
+
+#include "azure_iot_result.h"
+
+/**
+ * @brief Read the current trust bundle from the device storage.
+ *
+ * @param[in] pucTrustBundle The buffer into which the trust bundle will be copied.
+ * @param[in] ulTrustBundleLength The size of \p pucTrustBundle.
+ * @param[out] pulOutTrustBundleLength The number of bytes copied into \p pucTrustBundle.
+ * @param[in] pucTrustBundleVersion The buffer into which the trust bundle version will be copied.
+ * @param[in] ulTrustBundleVersionLength The size of \p pucTrustBundleVersion.
+ * @param[out] pulOutTrustBundleVersionLength The number of bytes copied into \p pucTrustBundleVersion.
+ * @return AzureIoTResult_t
+ */
+AzureIoTResult_t AzureIoTCAStorage_ReadTrustBundle( const uint8_t * pucTrustBundle,
+                                                    uint32_t ulTrustBundleLength,
+                                                    uint32_t * pulOutTrustBundleLength,
+                                                    const uint8_t * pucTrustBundleVersion,
+                                                    uint32_t ulTrustBundleVersionLength,
+                                                    uint32_t * pulOutTrustBundleVersionLength );
+
+/**
+ * @brief Write a trust bundle to the device storage.
+ *
+ * @param[in] pucTrustBundle The buffer containing the trust bundle.
```

Comment:
Could we add a bit more information about the format (i.e. PEM certificate store or concatenated PEM certificate files)?

Reply:
We could add for what we send in our implementation (aka in our Azure Function instructions), but because it's created and sent by the user, it could really be whatever they want. So it's not guaranteed to be PEM. I can add a note that it'll be in whatever format is sent from the custom allocation policy in the Azure Function?